### DISCUSSION TITLE
Power "on" via MQTT for x minutes and return to timer power state after that time

### DISCUSSION THREAD

**Discussion by mellow5** [arendst/Tasmota#13915] [Q&A] [answered]
I'm switching on a relay via MQTT message from node red for 15 min. After 15 minutes node red switches off again. I want to change the bahaviour in a way, that the relay returns to the intended state from the timer rule of tasmota after 15 minutes.

Any suggestions?

**Comment by sfromis** [comment, marked as answer]
You could automate that, using Tasmota rules, or of course in Node-Red. Pick you poison. Both would entail saving the desired power state, to restore it after expiry of the 15-minute interval.

**Comment by barbudor** [comment]
I would do it this way.
On Tasmota timers, do not trigger power directly but go through rule where in a backlog you save the intended power state in a var and do the power on/off
Have another event rule to be used from mqtt where you would power on, start a rule timer for 15 min. At the end of the 15 minutes you would set the power to the state saved in a var.
